resources/js/components/Elements/Slider.vue

```diff
@@ -17,23 +21,38 @@
             vertical: {
                 type: Boolean,
                 default: false
-            }
+            },
+            interval: {
+                type: Number,
+                default: 5,
+            },
+            autoplay: {
+                type: Boolean,
+                default: false,
+            },
         },
         data: () => {
             return {
                 position: 0,
                 showLeft: false,
                 showRight: false,
-                mounted: false
+                mounted: false,
+                hover: false
             }
         },
         mounted() {
             this.slider.addEventListener('scroll', this.scroll)
             this.slider.dispatchEvent(new CustomEvent('scroll'))
             this.mounted = true
+            if (this.autoplay && this.interval > 0) {
+                window.setInterval(this.autoScroll, this.interval * 1000)
```

Comment:
You can save yourself some headaches with the destroy by using
https://vueuse.org/shared/useIntervalFn/ (Same with the eventlistener and https://vueuse.org/core/useEventListener/)

It will automatically clear the interval if it is destroyed in vue
https://github.com/vueuse/vueuse/blob/main/packages/shared/useIntervalFn/index.ts#L73

And even returns controls so you could pause the scrolling if someone manually tried to scroll, add a pause button, you name it